packages/datadog-plugin-apollo-gateway/src/execute.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const TracingPlugin = require('../../dd-trace/src/plugins/tracing')
+const { storage } = require('../../datadog-core')
+
+class ApolloGatewayExecutePlugin extends TracingPlugin {
+  static get id () { return 'apollo-gateway' }
+  static get operation () { return 'execute' }
+  static get type () { return 'apollo-gateway' }
+
+  static get prefix () {
+    return 'tracing:apm:apollo-gateway:execute'
+  }
+
+  bindStart (ctx) {
+    const store = storage.getStore()
+    const childOf = store ? store.span : null
+
+    const span = this.startSpan(`${this.constructor.id}.${this.constructor.operation}`, {
+      childOf,
+      service: this.config.service,
+      type: this.constructor.type,
+      meta: {}
+    }, false)
+
+    ctx.parentStore = store
+    ctx.currentStore = { ...store, span }
+
+    return ctx.currentStore
+  }
+
+  asyncStart (ctx) {
+    ctx.currentStore.span.finish()
+    return ctx.parentStore
+  }
+
+  error (ctx) {
+    ctx.currentStore.span.setTag('error', ctx.error)
```

Comment:
Similar to above, this could fail.

Reply:
Also, why is this even needed given the general plugin also handles errors?

Reply:
just incase there is an error thrown at some point during the execution of the function that is not caught by apollo gateway. General plugin is only for errors that are explicitly caught by apollo gateway and then sent to the recordException method